esp-riscv-rt/src/lib.rs

```diff
@@ -543,8 +710,25 @@ _start_trap:
     sw s0, 30*4(sp)
 
     add a0, sp, zero
+    "#,
+    #[cfg(not(feature="direct-vectoring"))]
+    r#"
     jal ra, _start_trap_rust_hal
-
+    "#,
+    #[cfg(feature="direct-vectoring")] //jump to handler loaded in direct handler
+    r#"
+    addi sp, sp, -4 #build stack
+    sw ra, 0(sp)
+    jal ra, _handle_priority
```

Comment:
Lets put `_handle_priority` and `_restore_priority` behind an `interrupt-preemption` feature to match esp-hal.

Reply:
Is removing the interrupt-preemption related calls to these functions from esp-hal-common, and letting this routine handle it instead what you mean?

Reply:
I think it should work without changes and does make a lot of sense.

Reply:
As far as I understand it, it's possible to use direct vectoring with or without interrupt preemption, right now the interrupt process always tries to save/restore the priority level. We should cfg this away unless the feature is passed.

Reply:
Alright, now i'm on the same page. This would require an interrupt-preemption flag to be added to the RT, so maybe we might as well move the interrupt-preemption code out of esp-hal-common? It is sort of more runtime related anyway...

Reply:
That might be for another PR though, for now i'll add ``#cfg[all(feature="direct-vectoring", feature="interrupt-preemption")] here.